crypto/cms/cms_env.c

```diff
@@ -267,6 +267,7 @@ int CMS_RecipientInfo_ktri_get0_signer_id(CMS_RecipientInfo *ri,
     return cms_SignerIdentifier_get0_signer_id(ktri->rid, keyid, issuer, sno);
 }
 
+/* -1 is a valid return value, so we use -2 for error here */
```

Comment:
Why not just return `OSSL_E_UNSUPPORTED`? Is this name really unapt in this context? If it is, you can figure out a different name ([look here](https://docs.microsoft.com/en-us/windows/win32/com/com-error-codes-1) for inspirations). Returning a different negative value, because the return value is not properly documented anyway ;-)

```
=head1 RETURN VALUES

...

CMS_RecipientInfo_ktri_cert_cmp() and CMS_RecipientInfo_kekri_cmp() return 0
for a successful comparison and non zero otherwise.
```

Reply:
The comparison result is clearly modelled after the likes of strcmp, even though the documentation is crap. If we change the return values, we risk breaking applications unnecessarily.